test/e2e/atlas/compliancepolicy_copyprotection_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e || (atlas && backup && compliancepolicy)
+
+package atlas_test
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+//nolint:thelper
+func testCopyProtection(t *testing.T, g *atlasE2ETestGenerator) {
+	cliPath, err := e2e.AtlasCLIBin()
+	r := require.New(t)
+	r.NoError(err)
+
+	t.Run("copyprotection happy flow", func(t *testing.T) {
+		cmd := exec.Command(
+			cliPath,
+			backupsEntity,
+			compliancepolicyEntity,
+			"copyprotection",
+			"enable",
+			"-o=json",
+			"--projectId",
+			g.projectID,
+			"--watch",
+		)
+		cmd.Env = os.Environ()
+		resp, outputErr := cmd.CombinedOutput()
+		r.NoError(outputErr, string(resp))
+
+		trimmedResponse := removeDotsFromWatching(resp)
+
+		a := assert.New(t)
+
+		var compliancepolicy atlasv2.DataProtectionSettings
+		require.NoError(t, json.Unmarshal(trimmedResponse, &compliancepolicy), string(trimmedResponse))
+		a.True(*compliancepolicy.CopyProtectionEnabled)
+	})
+
+	t.Run("copyprotection invalid argument", func(t *testing.T) {
+		invalidArgument := "invalid"
+		cmd := exec.Command(
+			cliPath,
+			backupsEntity,
+			compliancepolicyEntity,
+			"copyprotection",
+			invalidArgument)
+		cmd.Env = os.Environ()
+		_, err = cmd.CombinedOutput()
+		r.Error(err)
+	})
```

Comment:
what do we gain with this test? this is the kind of test that would be better fitted for a unit tests. Also to keep in mind this command should be refactored as previously discussed to be two sub commands instead of arguments

Reply:
My intention was to implement a sad flow test, but I agree, this is very "unity'. Do you have any pointers/suggestions/guidelines as to what kind of E2E sad flow tests to implement generally?
For this particular test, I will remove it when I refactor this command and test the behaviour with unit tests instead.

Reply:
>pointers/suggestions/guidelines

E2E are slow, ideally you want to follow the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html), you e2e will usually be happy path or mission critical

This test in particular is testing the framework (cobra) to make sure the framework work as expected, we shouldn't test the framework, even less when the option to pass "any" argument is removed